### PR TITLE
nfc: add library variants for nRF53

### PR DESCRIPTION
**Pull request by kapi-no**
Added NFC T2T and T4T libraries for nRF5340 SoC.

Signed-off-by: Kamil Piszczek <Kamil.Piszczek@nordicsemi.no>

### CODE REVIEW

**Comment by kapi-no** [comment]
@SebastianBoe, please take a look at Cmake related changes

**Comment by kapi-no** [comment]
> 
> 
> Update CHANGELOG.rst, since addition of M33-compiled version should be mentioned there. Also #115 should be reflected there, but was forgotten.

This is updated now

File: nfc/CMakeLists.txt [hunk 1]
```diff
@@ -18,13 +18,27 @@ endif()
...
 endif()
 
 if(CONFIG_NFC_T4T_NRFXLIB)
-	zephyr_link_libraries(${NFC_LIB_PATH}/libnfct4t_nrf52.a)
+	if (CONFIG_SOC_SERIES_NRF52X)
+		zephyr_link_libraries(${NFC_LIB_PATH}/libnfct4t_nrf52.a)
+	endif()
+	if (CONFIG_SOC_SERIES_NRF53X)
+		zephyr_link_libraries(${NFC_LIB_PATH}/libnfct4t_nrf53.a)
+	endif()
```
Comment (SebastianBoe) [inline nfc/CMakeLists.txt]:
```
if(CONFIG_NFC_T2T_NRFXLIB)
  zephyr_link_libraries(${NFC_LIB_PATH}/libnfct2t_${CONFIG_SOC_SERIES}.a)
endif()

```

Comment (kapi-no) [reply nfc/CMakeLists.txt]: Done, thanks for the tip, it looks cleaner now.

**Comment by SebastianBoe** [comment]
DNM until the upmerge procedure is complete

**Comment by kapi-no** [comment]
@b-gent or @greg-fer, could you review Changelog?

File: nfc/CHANGELOG.rst [hunk 2]
```diff
@@ -5,6 +5,20 @@ Changelog
...
+NFC 0.4.0
+****************
+
+Added NFC Type 2 Tag and Type 4 Tag libraries for nRF5340 SoC that are
+compiled for Cortex-M33 CPU in hard and soft float variants.
+
+Added
+=====
+
+* Added the NFC T2T library (``libnfct2t_nrf53.a``) for nRF5340 SoC.
```
Comment (greg-fer) [inline nfc/CHANGELOG.rst]:
```suggestion
* Added the NFC T2T library for the nRF5340 device: ``libnfct2t_nrf53.a``.
```

File: nfc/CHANGELOG.rst [hunk 3]
```diff
@@ -5,6 +5,20 @@ Changelog
...
+* Added the NFC T4T library (``libnfct4t_nrf53.a``) for nRF5340 SoC.
```
Comment (greg-fer) [inline nfc/CHANGELOG.rst]:
```suggestion
* Added the NFC T4T library for the nRF5340 device: ``libnfct4t_nrf53.a``.
```

File: nfc/CHANGELOG.rst [hunk 4]
```diff
@@ -5,6 +5,20 @@ Changelog
...
+* Added feature to the NFC T4T library that allows the user to change NDEF
```
Comment (greg-fer) [inline nfc/CHANGELOG.rst]:
```suggestion
* Added a feature to the NFC T4T library that allows the user to change the NDEF
```

File: nfc/CHANGELOG.rst [hunk 5]
```diff
@@ -5,6 +5,20 @@ Changelog
...
+  buffer when NFC stack is already initialized and running.
```
Comment (greg-fer) [inline nfc/CHANGELOG.rst]:
```suggestion
  buffer when the NFC stack is already initialized and running.
```

File: nfc/CHANGELOG.rst [hunk 6]
```diff
@@ -5,6 +5,20 @@ Changelog
 
 All notable changes to this project are documented in this file.
 
+NFC 0.4.0
+****************
+
+Added NFC Type 2 Tag and Type 4 Tag libraries for nRF5340 SoC that are
```
Comment (greg-fer) [inline nfc/CHANGELOG.rst]:
```suggestion
Added Type 2 Tag and Type 4 Tag NFC libraries for the nRF5340 device.
```

File: nfc/CHANGELOG.rst [hunk 7]
```diff
@@ -5,6 +5,20 @@ Changelog
...
+compiled for Cortex-M33 CPU in hard and soft float variants.
```
Comment (greg-fer) [inline nfc/CHANGELOG.rst]:
```suggestion
Both libraries are compiled for the Cortex-M33 CPU in hard and soft float variants.
```

**Comment by kapi-no** [comment]
@greg-fer, thanks for the review. I addressed your comments

**Comment by kapi-no** [comment]
@SebastianBoe, could you remove DNM label since the Zephyr upmerge is finished?

**Comment by tejlmand** [comment]
@grochu and @kapi-no  Why doesn't this PR has a corresponding manifest update in https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/master/west.yml ?

It means this PR is not fully tested in CI.

**Comment by kapi-no** [comment]
@tejlmand, well, we didn't need the changes yet because we are waiting for the next Zephyr upmerge. I will open PR to nrf repository to verify if CI is okay with this change.

**Comment by tejlmand** [comment]
@kapi-no understood. But in that case it might be better to wait with merging the PR, as other PR's following this one will the carry along the updated NFC library, as example: https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1693#issuecomment-571120102

**Pull request merged**